### ISSUE TITLE
fix: LearningRateMonitor(logging_interval='epoch') logs per epoch instead of step

### ISSUE THREAD

**Issue by ouioui199** [PyTorchLightning/pytorch-lightning#19275]
### Outline & Motivation

I am new to lightning, for the last few days I have been trying to convert my pytorch code into lightning. 
I want to use Tensorboard to visualize training progression, including the learning rate.
I see that when argument "logging_interval='epoch'" has been set, the LearningRateMonitor still logs with step.

### Pitch

Change "step=trainer.fit_loop.epoch_loop._batches_that_stepped" to "step=trainer.current_epoch" in line 180 on file lightning/pytorch/callbacks/lr_monitor.py

cc @lantiga @justusschock @awaelchli

**Comment by SkafteNicki** [comment]
This is working as intended. Setting `logging_interval="epoch"` does not mean to log the learning rate as a function of the epoch number, it means that the learning rate will be logged when a epoch end but still use the global step as the step value. This is similar to `self.log` will always log using `step=self.global_step` even if it is called inside a `on_*_epoch_end` hook. This is to keep a consistent logging level across all of lightning. However, lightning does automatically log a `(step, epoch)` curve as training progresses that can be used to convert any other logged value into a plot with `epoch` on the x-axis.
Closing issue, but feel free to ping and reopen if necessary.
